include/linux/kftf.h

```diff
@@ -125,4 +126,101 @@ static int write_input_cb_common(struct file *filp, const char __user *buf,
 	}                                                                      \
 	static void _fuzz_test_logic_##func(func_arg_type arg)
 
+/**
+ * Reports a bug with a predictable prefix so that it can be parsed by a
+ * fuzzing driver.
+ */
+#define KFTF_REPORT_BUG(msg, fmt) pr_warn("bug: " #msg, fmt)
+
+/**
+ * struct kftf_constraint_type defines a type of constraint. The fuzzing driver
+ * should be aware of these.
+ */
+enum kftf_constraint_type : uint8_t {
+	EXPECT_EQ = 0,
+	EXPECT_NE,
+	EXPECT_LE,
+	EXPECT_GT,
+	EXPECT_IN_RANGE,
+};
+
+/**
+ * ktft_constraint defines a domain constraint for a struct variable that is
+ * taken as input for a FUZZ_TEST
+ *
+ * @input_type: the name of the input (a struct name)
+ * @field_name: the name of the field that this domain constraint applies to
+ * @value1: used in all comparisons
+ * @value2: only used in comparisons that require multiple values, e.g. range
+ *	constraints
+ * @type: the type of the constraint, enumerated above
+ *
+ * Note: if this struct is not a multiple of 64 bytes, everything breaks and
+ * we get corrupted data and occasional kernel panics. To avoid this happening,
+ * we enforce 64 Byte alignment and statically assert that this struct has size
```

Comment:
I am afraid we'll have bigger structs in the future (e.g. to list a set of possible values).
Why does everything break?
Maybe we can put the struct length at its beginning?